# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxUserFontSet.h
# Commit: 5a39eb924188
# Full Hash: 5a39eb924188e504dad274a2148543c8d51dbae8
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2022-06-29 03:44:30
# Regressor Bug: 1771493
# File Overlap Count: 1
# Description:
#   Bug 1771493 - Part 2. Merge FontFaceSetImpl and gfxUserFontSet into a single class. r=emilio
#   
#   This patch merges FontFaceSetImpl and gfxUserFontSet into the same
#   class. This reduces the level of indirection and in theory makes it
#   easier to manage future threading issues.
# ==============================================================================

diff -r 3d30a8d83d77 -r 5a39eb924188 gfx/thebes/gfxUserFontSet.h
--- a/gfx/thebes/gfxUserFontSet.h	Tue Jun 28 21:46:59 2022 +0000
+++ b/gfx/thebes/gfxUserFontSet.h	Tue Jun 28 21:47:00 2022 +0000
@@ -91,7 +91,8 @@
 
   // The principal that should be used for the load. Should only be used for
   // URL sources.
-  gfxFontSrcPrincipal* LoadPrincipal(const gfxUserFontSet&) const;
+  already_AddRefed<gfxFontSrcPrincipal> LoadPrincipal(
+      const gfxUserFontSet&) const;
 };
 
 inline bool operator==(const gfxFontFaceSrc& a, const gfxFontFaceSrc& b) {
@@ -225,10 +226,12 @@
   typedef mozilla::WeightRange WeightRange;
   typedef gfxFontEntry::RangeFlags RangeFlags;
 
-  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(gfxUserFontSet)
+  NS_INLINE_DECL_PURE_VIRTUAL_REFCOUNTING
 
   gfxUserFontSet();
 
+  void Destroy();
+
   enum {
     // no flags ==> no hint set
     // unknown ==> unknown format hint set
@@ -300,17 +303,13 @@
   // the given name
   gfxUserFontFamily* LookupFamily(const nsACString& aName) const;
 
-  virtual gfxFontSrcPrincipal* GetStandardFontLoadPrincipal() const = 0;
+  virtual already_AddRefed<gfxFontSrcPrincipal> GetStandardFontLoadPrincipal()
+      const = 0;
   virtual nsPresContext* GetPresContext() const = 0;
 
   // check whether content policies allow the given URI to load.
   virtual bool IsFontLoadAllowed(const gfxFontFaceSrc&) = 0;
 
-  // Dispatches all of the specified runnables to the font face set's
-  // document's event queue.
-  virtual void DispatchFontLoadViolations(
-      nsTArray<nsCOMPtr<nsIRunnable>>& aViolations) = 0;
-
   // initialize the process that loads external font data, which upon
   // completion will call FontDataDownloadComplete method
   virtual nsresult StartLoad(gfxUserFontEntry* aUserFontEntry,